app/src/main/cpp/skyline/services/codec/IHardwareOpusDecoder.h

```diff
@@ -0,0 +1,110 @@
+
+    static constexpr u32 OpusFullbandSampleRate{48000};
+    static constexpr u32 MaxFrameSizeNormal = OpusFullbandSampleRate * 0.040; //!< 40ms frame size limit for normal decoders
+    static constexpr u32 MaxFrameSizeMultiEx = OpusFullbandSampleRate * 0.120; //!< 120ms frame size limit for ex decoders added in 12.0.0
+    static constexpr u32 MaxInputBufferSize{0x600}; //!< Maximum allocated size of the input buffer
+
+    /**
+     * @brief This service decodes an Opus audio stream. Always created with OpenHardwareOpusDecoder by IHardwareOpusDecoderManager
+     * @url https://switchbrew.org/wiki/Audio_services#IHardwareOpusDecoder
+     */
+    class IHardwareOpusDecoder : public BaseService {
+      private:
+        std::shared_ptr<kernel::type::KTransferMemory> workBuffer;
+        OpusDecoder *decoderState{};
+        i32 sampleRate;
+        i32 channelCount;
+        i32 decoderOutputBufferSize;
+
+        /**
+         * @brief Resets the codec state to be equivalent to a freshly initialized state
+         */
+        void ResetContext();
+
+        /**
+         * @brief Called by all DecodeInterleaved* functions
+         * @param performanceInfo Whether to return decode time taken or not
+         */
+        Result DecodeInterleavedImpl(ipc::IpcRequest &request, ipc::IpcResponse &response, bool performanceInfo = false);
+
+      public:
+        /**
+         * @param sampleRate The sample rate of the Opus audio data
+         * @param channelCount The channel count of the Opus audio data
+         */
+        IHardwareOpusDecoder(const DeviceState &state, ServiceManager &manager, i32 sampleRate, i32 channelCount, u32 workBufferSize, KHandle kWorkBuffer);
+
+        /**
+         * @brief Takes an OpusDataIn input buffer and a PcmDataOut output buffer.
+         * Decodes the Opus source data to the output buffer and returns decoded data size and decoded sample count
+         * @url https://switchbrew.org/wiki/Audio_services#DecodeInterleavedOld
+         */
+        Result DecodeInterleavedOld(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response);
+
+        /**
+         * @brief Takes an OpusDataIn input buffer and a PcmDataOut output buffer.
+         * Decodes the Opus source data to the output buffer and returns decoded data size, decoded sample count and decode time taken in microseconds
+         * @url https://switchbrew.org/wiki/Audio_services#DecodeInterleavedWithPerfOld
+         */
+        Result DecodeInterleavedWithPerfOld(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response);
+
+        /**
+         * @brief Same as DecodeInterleaved
+         * @see DecodeInterleaved
+         * @url https://switchbrew.org/wiki/Audio_services#DecodeInterleaved
+         */
+        Result DecodeInterleavedWithPerfAndResetOld(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response);
```

Comment:
Repetition of the information regarding it being the same is extremely redundant, we won't ever generate Doxygen documentation for Skyline so it's really pointless to have stuff like `@see`. Additionally, don't link to Switchbrew when there isn't a link that directly corresponds to that specific function.

The information covered here in itself is just redundant if it is `Same as DecodeInterleaved` because the point of it would be to cover the differences but if there aren't any then the function in itself becomes redundant and you could just eliminate it and use `SFUNC(0x6, IHardwareOpusDecoder, DecodeInterleaved)` with a comment there `// DecodeInterleavedWithPerfAndResetOld is effectively the same as DecodeInterleaved`.